src_app.py

```diff
@@ -1,41 +1,139 @@
+
+    # return the model
+    return linear_regression_model
+
+
+def predict(linear_regression_model, house_features):
+    """predinc the house price using the linear regression model for the house
+       features
+
+    Args:
+        linear_regression_model ( sklearn.linear_model._base.LinearRegression):
+            the linear regression model
+        house_features (dict): the house features to predict for
+
+    Returns:
+        dict: the predict value stored in dict
+    """
+
+    # slice the features values
+    features_values = [float(house_features.get(
+        feature)) for feature in config.FEATURES_NAMES_IN_ORDER]
+
+    # create a numpy array of the features values and reshape it into 2D,
+    # due to he predict function assumes a 2D array where each line is a
+    # sample to predict for
+    features_values_in_shape = np.array(features_values).reshape(1, 6)
+
+    # make the prediction
+    predicted_house_price = linear_regression_model.predict(
+        features_values_in_shape)
+
+    # slice the predicted value
+    json_reslut = {config.PREDICTED_VALUE: predicted_house_price[0]}
+
+    # return the predicted value
+    return json_reslut
+
+
 @app.route('/')
-def hello():
+def welcoming():
     return render_template('index.html')
 
 
 @app.route('/predict', methods=['GET', 'POST'])
 def predict_house_price():
 
+    # if the method is POST
     if request.method == 'POST':
 
-        # import the predicting module
-
-        import linear_model
-
         # get the data from the POST request
-
         house_features = request.get_json()
 
         # call the predicting function and return the result
+        return json.dumps(predict(linear_regression_model, house_features))
 
-        return linear_model.predict(house_features, parameters)
+    # if the method is GET
     else:
 
-        return render_template('index.html')
+        # get the house_features from the get request
+        house_features = {
+            feature: request.args.get(feature)
+            for feature in config.FEATURES_NAMES_IN_ORDER
+        }
+
+        # call the predicting function and return the result
+        return json.dumps(predict(linear_regression_model, house_features))
+
 
 if __name__ == '__main__':
 
-    # read the parameters(thetas and theta0) from json file
+    # get the model file path from the console
+    pickle_file_path = read_arguments()
+
+    # check if the file exists or not
+    if not os.path.isfile(pickle_file_path):
+
+        # display an error
+        print("Error. Please make sure the file is existed!")
+
+    else:
 
-    with open('lm_parameters.json', 'r') as parameters_file:
-        parameters = json.loads(parameters_file.read())
+        # read the linear regression model
+        linear_regression_model = read_linear_regression_model(
+            pickle_file_path)
 
-    app.run(debug=True, use_reloader=True)
+        # run the service
+        app.run(debug=True, use_reloader=True)
```

Comment:
it's better to declare the port in the app.run when you're working with requests for production purposes.